app/scripts/controllers/transactions/pending-tx-tracker.js

```diff
@@ -145,16 +145,16 @@ class PendingTransactionTracker extends EventEmitter {
     // If another tx with the same nonce is mined, set as failed.
     const taken = await this._checkIfNonceIsTaken(txMeta)
     if (taken) {
-      const nonceTakenErr = new Error('Another transaction with this nonce has been mined.')
-      nonceTakenErr.name = 'NonceTakenErr'
-      return this.emit('tx:failed', txId, nonceTakenErr)
+      return this.emit('tx:dropped', txId)
```

Comment:
any reason to not use setTxStatusDropped here?

Reply:
unify behavior and expectations

Reply:
setTxStatusDropped emits status update events, not pendingTx events